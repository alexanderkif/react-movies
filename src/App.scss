@import './styles/colors.scss';

.App {
  display:flex;
  flex-direction:column;
  max-width: 1024px;
  min-height: 100vh;
  margin: 0 auto;
  color: $text;
}
