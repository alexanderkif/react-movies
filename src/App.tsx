import React, { FunctionComponent } from "react";
import styles from "./App.scss";
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { createStore, applyMiddleware, compose } from "redux";
import { rootReducer } from "./redux/reducers";
import Search from "./pages/Search";
import Details from "./pages/Details";
import Footer from "./components/Footer";
import Page404 from "./pages/Page404";

const composeEnhancers =
  // eslint-disable-next-line
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
export const store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(thunk))
);

export const App: FunctionComponent = () => {
  return (
    <div className={styles.container}>
      <Provider store={store}>
        <Router>
          <Switch>
            <Route path="/movies/:id">
              <Details />
            </Route>
            <Route exact path="/">
              <Search />
            </Route>
            <Route component={Page404} />
          </Switch>
          <Footer />
        </Router>
      </Provider>
    </div>
  );
};
