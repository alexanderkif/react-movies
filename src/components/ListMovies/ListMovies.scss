@import '../../styles/colors.scss';

.ListMovies {
  display: flex;
  flex-wrap: wrap;
  gap: 3.8rem 3.5rem;
  padding: 4.3rem 3rem;
  min-height: 35.5rem;
  background-color: $dark;
  color: $light;
}
