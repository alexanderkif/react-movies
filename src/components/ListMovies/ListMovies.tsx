import React, { FunctionComponent } from "react";
import styles from "./ListMovies.scss";
import Movie from "../../pages/Movie";
import { IListMoviesProps } from "../../types";

export const ListMovies: FunctionComponent<IListMoviesProps> = (
  props: IListMoviesProps
) => {
  const { movies } = props;

  return (
    <div className={styles.wrapper}>
      {movies?.map((movie) => (
        <Movie key={movie.id} movie={movie} />
      ))}
    </div>
  );
};
