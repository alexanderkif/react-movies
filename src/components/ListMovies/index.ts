import { ListMoviesComponent } from './ListMovies'

export default ListMoviesComponent
