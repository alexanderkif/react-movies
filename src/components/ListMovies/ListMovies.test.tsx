/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import ListMovies from '.';
import { store } from '../../App';

const movies = [
  {
    id: 447365,
    title: 'Guardians of the Galaxy Vol. 3',
    tagline: '',
    vote_average: 0,
    vote_count: 9,
    release_date: '2020-05-01',
    poster_path:
      'https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg',
    overview: "The third film based on Marvel's Guardians of the Galaxy.",
    budget: 0,
    revenue: 0,
    genres: ['Action', 'Adventure', 'Science Fiction'],
    runtime: 5,
  },
  {
    id: 181790,
    title: 'Journey 3: From the Earth to the Moon',
    tagline: '',
    vote_average: 0,
    vote_count: 7,
    release_date: '2018-12-31',
    poster_path:
      'https://image.tmdb.org/t/p/w500/98tbNloMBztAVnWpAznKKVUdi2O.jpg',
    overview: 'Sean and Hank go on their biggest adventure yet, to the moon.',
    budget: 0,
    revenue: 0,
    genres: ['Action', 'Adventure', 'Fantasy', 'Science Fiction'],
    runtime: 5,
  },
];

describe('ListMovies', () => {
  it('snapshot ListMovies', () => {
    const element = render(
      <Provider store={store}>
        <ListMovies movies={movies} />
      </Provider>,
    );
    expect(element).toMatchSnapshot();
  });
});
