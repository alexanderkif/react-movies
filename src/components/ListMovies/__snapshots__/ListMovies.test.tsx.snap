// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMovies snapshot ListMovies 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="card"
        >
          <img
            class="picture"
            src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
          />
          <div
            class="edit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </div>
          <div
            class="delete"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </div>
          <div
            class="nameYear"
          >
            <div
              class="name"
            >
              Guardians of the Galaxy Vol. 3
            </div>
            <div
              class="year"
            >
              2020
            </div>
          </div>
          <div
            class="genre"
          >
            Action & Adventure & Science Fiction
          </div>
        </div>
        <div
          class="card"
        >
          <img
            class="picture"
            src="https://image.tmdb.org/t/p/w500/98tbNloMBztAVnWpAznKKVUdi2O.jpg"
          />
          <div
            class="edit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </div>
          <div
            class="delete"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </div>
          <div
            class="nameYear"
          >
            <div
              class="name"
            >
              Journey 3: From the Earth to the Moon
            </div>
            <div
              class="year"
            >
              2018
            </div>
          </div>
          <div
            class="genre"
          >
            Action & Adventure & Fantasy & Science Fiction
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="card"
      >
        <img
          class="picture"
          src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
        />
        <div
          class="edit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
        <div
          class="delete"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </div>
        <div
          class="nameYear"
        >
          <div
            class="name"
          >
            Guardians of the Galaxy Vol. 3
          </div>
          <div
            class="year"
          >
            2020
          </div>
        </div>
        <div
          class="genre"
        >
          Action & Adventure & Science Fiction
        </div>
      </div>
      <div
        class="card"
      >
        <img
          class="picture"
          src="https://image.tmdb.org/t/p/w500/98tbNloMBztAVnWpAznKKVUdi2O.jpg"
        />
        <div
          class="edit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
        <div
          class="delete"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </div>
        <div
          class="nameYear"
        >
          <div
            class="name"
          >
            Journey 3: From the Earth to the Moon
          </div>
          <div
            class="year"
          >
            2018
          </div>
        </div>
        <div
          class="genre"
        >
          Action & Adventure & Fantasy & Science Fiction
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
